### PR TITLE
Update Symbol signatures

### PR DESCRIPTION
**Pull request by hanachin**

### CODE REVIEW

**Comment by hanachin** [comment]
Now it's ready for review.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require: rubocop-rubycw
 AllCops:
-  TargetRubyVersion: 2.6
+  TargetRubyVersion: 2.7
```
Comment (soutaro) [inline .rubocop.yml]: 👍

File: stdlib/builtin/symbol.rbs [hunk 2]
```diff
@@ -36,99 +34,202 @@ class Symbol < Object
...
-  # ```
-  def to_proc: () -> Proc
+  #     :hello.start_with?("hell")               #=> true
+  #     :hello.start_with?(/H/i)                 #=> true
+  # 
+  #     # returns true if one of the prefixes matches.
+  #     :hello.start_with?("heaven", "hell")     #=> true
+  #     :hello.start_with?("heaven", "paradise") #=> false
+  # 
+  # variadic type parameter is not supported yet
```
Comment (soutaro) [inline stdlib/builtin/symbol.rbs]: What does this comment mean?? @hanachin 
I think the type is the best one for this method.

File: stdlib/builtin/symbol.rbs [hunk 3]
```diff
@@ -36,99 +34,202 @@ class Symbol < Object
...
+  # 
+  def to_proc: () -> Proc
 
-  # Returns the name or string corresponding to *sym* .
+  # Returns the name or string corresponding to *sym*.
+  # 
+  #     :fred.id2name   #=> "fred"
+  #     :ginger.to_s    #=> "ginger"
+  # 
+  # # arglists 💪👽🚨 << Delete this section
```
Comment (soutaro) [inline stdlib/builtin/symbol.rbs]: 💪 👽 🚨

File: stdlib/builtin/symbol.rbs (hunk 2 above)
Comment (hanachin) [reply stdlib/builtin/symbol.rbs]: Woops, I forget to delete that.
First I wrote `def start_with?: (String str) -> bool` then realize this method take more than one arguments, but I didn't get how to write variadic arguments at the moment. Finally I got it but forgot to :scissors: the comments.

File: stdlib/builtin/symbol.rbs (hunk 3 above)
Comment (hanachin) [reply stdlib/builtin/symbol.rbs]: :see_no_evil: 

 I added a test and update the signature and remove the comment.

**Pull request merged**